packages/vite/src/node/utils.ts

```diff
@@ -642,31 +648,31 @@ function windowsMappedRealpathSync(path: string) {
   return realPath
 }
 const parseNetUseRE = /^(\w+)? +(\w:) +([^ ]+)\s/
-let firstSafeRealPathSyncRun = false
+let firstSafeRealPathRun = false
 
-function windowsSafeRealPathSync(path: string): string {
-  if (!firstSafeRealPathSyncRun) {
-    optimizeSafeRealPathSync()
-    firstSafeRealPathSyncRun = true
+async function windowsSafeRealPath(path: string): Promise<string> {
+  if (!firstSafeRealPathRun) {
+    await optimizeSafeRealPath()
+    firstSafeRealPathRun = true
   }
   return fs.realpathSync(path)
 }
 
-function optimizeSafeRealPathSync() {
+async function optimizeSafeRealPath() {
   // Skip if using Node <18.10 due to MAX_PATH issue: https://github.com/vitejs/vite/issues/12931
   const nodeVersion = process.versions.node.split('.').map(Number)
```

Comment:
Good idea! Lets do that in another PR 👍

Reply:
Oh no my bad this is actually called only once!

Reply:
Yes, I thought you wanted to start  the check when loading the module instead of in the first call

Reply:
No I think is better like this because done only once for Windows users instead of at module load for everyone.
I though this was called multiple times

Reply:
We could only do the init for windows only. But ya, not much to gain